### PR TITLE
Cdnc 1781 v3. Adding an API that allows a simple restart or terminate then restart if there is an active runID

### PR DESCRIPTION
**Pull request by allenchen2244**
IDL for restart workflow api

### CODE REVIEW

File: proto/uber/cadence/api/v1/service_workflow.proto
Comment (davidporter-id-au) [inline proto/uber/cadence/api/v1/service_workflow.proto:99]: What is identity?

File: thrift/cadence.thrift
Comment (davidporter-id-au) [inline thrift/cadence.thrift:103]: I think `2:` might be missing here. But overall, I think these are all 4XX errors, nice!

**Comment by davidporter-id-au** [comment]
Please include more information in the title and description about what you're trying to achieve for future folks, particularly given the risk of ambiguity with this API

File: proto/uber/cadence/api/v1/service_workflow.proto
Comment (allenchen2244) [reply proto/uber/cadence/api/v1/service_workflow.proto:99]: identity is the guy making the call. it gets passed in in all requests pretty much

File: thrift/cadence.thrift
Comment (allenchen2244) [reply thrift/cadence.thrift:103]: woops will fix

File: proto/uber/cadence/api/v1/service_workflow.proto
Comment (davidporter-id-au) [reply proto/uber/cadence/api/v1/service_workflow.proto:99]: ah, sorry, I was expecting AuthN information to be passed as a header. But, as you say, this seems to be a field throughout the rest of the APIs. ok, fair.

**Review by davidporter-id-au — APPROVED** [review]
lgtm, maybe if we could get a second pair of eyes that might be good, just because IDL is one of those things that is tricky to change later on.

File: thrift/cadence.thrift
Comment (Groxx) [inline thrift/cadence.thrift:100]: using `Start*`, not `Restart*` response

File: proto/uber/cadence/api/v1/service_workflow.proto
Comment (Groxx) [inline proto/uber/cadence/api/v1/service_workflow.proto:96]: if the semantics we're going for are always "current run is terminated", we probably need to support either request-ID or an "only if current" / "only if current == UUID" argument.

otherwise, with a successful-but-timed-out request, you risk restarting a *second* time, and not having a good way to prevent that.

Comment (allenchen2244) [reply proto/uber/cadence/api/v1/service_workflow.proto:96]: what does this mean? "with a successful-but-timed-out request, you risk restarting a second time, and not having a good way to prevent that."

Right now it calls startworkflow with the policy terminate if the wfid has something running.

**Pull request merged**
